core/corehttp/gateway_handler.go

```diff
@@ -316,9 +317,25 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 		}
 	}
 
-	contentPath := ipath.New(r.URL.Path)
+	redirects, err := i.searchUpTreeForRedirects(r, urlPath)
```

Comment:
@lidel should block/car requests continue to return 404s if not present?  I'm assuming so, and if true, I think we're going to have issues moving redirect logic into `gateway_handler_unixfs.go` since the 404 logic kicks in before we have a chance to read the `_redirects` file and we return 404 before having a chance to redirect.

I can think more about this but wanted to confirm I'm thinking clearly, since this code is quite new to me.  Thanks.

Reply:
@justincjohnson No, block/car responses should not have any special magic to them.  Apply 404 / _redirect magic only to default unixfs code paths. 

We want to keep (soon to be created) spec for raw/car responses minimal.
That will allow people to implement and run light "trustless transport gateways" that do not render unixfs (e.g. for IoT use cases).

Reply:
@lidel I pushed new changes to my branch that I believe generally address your feedback.  The branch is still WIP but it's working for me.  I'm new to Go and IPFS and would be very grateful for any feedback you're willing to give.  🙏